### DISCUSSION TITLE
Prefer $this return type over static

### DISCUSSION THREAD

**Discussion by dpi** [rectorphp/rector#7437] [Q&A] [answered]
Hi there, 

We have a codebase where there are many `@return $this`. We've run Rector with just a handful of return type rules* and found its adding `static` as a return type. We'd prefer to keep our methods return type-less until something like https://wiki.php.net/rfc/this_return_type comes along. I believe it may also have different semantics for PHPStan and code style tools as you can tell its the same object being chained.

Is there a recommendation for working with this scenario.

*The rules used are: `AddVoidReturnTypeWhereNoReturnRector`, `ReturnTypeFromStrictNativeCallRector`, `ReturnTypeFromStrictScalarReturnExprRector`, `ReturnTypeDeclarationRector`.

Thanks!

**Comment by TomasVotruba** [comment, marked as answer]
Hi,

the best way is to add rule that add what you don't want via `skip()` method:
https://github.com/rectorphp/rector/blob/main/docs/how_to_ignore_rule_or_paths.md

On the other hand, the `static` return type is far better than docblocks, as it give PHPStan and Rector 100 % confidence about type.
So more rule can work with it and build on it :)

**Reply by dpi** [reply]
Thanks @TomasVotruba , 

So there isn't a way to configure specific types? 

We need to disabling a rule entirely by path :/

**Reply by TomasVotruba** [reply]
It's not, as the point of these rules is to improve type declarations in your code :)
I'd not use Rector if you want to keep your code unchanged.
